webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala

```diff
@@ -1181,6 +1182,276 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
         } yield taskResult
     }
 
+    /**
+     * Changes position of the node
+     *
+     * @param nodeIri                   the node's IRI.
+     * @param changeNodePositionRequest the new node comments.
+     * @param featureFactoryConfig      the feature factory configuration.
+     * @param requestingUser            the requesting user.
+     * @param apiRequestID              the unique api request ID.
+     * @return a [[NodeInfoGetResponseADM]]
+     * @throws ForbiddenException          in the case that the user is not allowed to perform the operation.
+     * @throws UpdateNotPerformedException in the case something else went wrong, and the change could not be performed.
+     */
+    private def nodePositionChangeRequest(nodeIri: IRI,
+                                          changeNodePositionRequest: ChangeNodePositionApiRequestADM,
+                                          featureFactoryConfig: FeatureFactoryConfig,
+                                          requestingUser: UserADM,
+                                          apiRequestID: UUID): Future[ListADM] = {
+        /**
+         * Checks if the given position is in range.
+         * The highest position a node can be placed is to the end of the parents children; that means length of existing
+         * children + 1
+         *
+         * @param parentNode the parent to which the node should belong.
+         * @param isNewParent identifier that node is added to another parent or not.
+         *
+         * @throws BadRequestException if given position is out of range.
+         */
+        def isNewPositionValid(parentNode: ListNodeADM, isNewParent: Boolean): Unit = {
+            val numberOfChildren = parentNode.getChildren.size
+            // If the node must be added to a new parent, highest valid position is numberOfChildren+1
+            // That means the furthests a node can be positioned is being appended to the end of children of the new parent.
+            if(isNewParent && changeNodePositionRequest.position >  numberOfChildren + 1) {
+                throw BadRequestException(s"Invalid position given, maximum allowed is=${numberOfChildren + 1}!")
```

Comment:
Maybe this is just me, but I don't like receiving error messages with exclamation marks in them, it makes me feel like the computer is shouting at me. :)